source/parsing/Preprocessor_macros.cpp

```diff
@@ -59,6 +59,17 @@ void Preprocessor::createBuiltInMacro(string_view name, int value, string_view v
 MacroActualArgumentListSyntax* Preprocessor::handleTopLevelMacro(Token directive) {
```

Comment:
Probably you need to make this function return a std::pair<MacroActualArgumentListSyntax*, Trivia> and return the skipped tokens as a Trivia(TriviaKind::SkippedTokens...). handleMacroUsage then gets changed to return a std::pair<Trivia, Trivia> and returns this extra trivia result as the second member, and then the usage inside handleDirective checks for that trivia and adds it to the list if needed.